### ISSUE TITLE
Add ability to configure preferred sort order of shelves

### ISSUE THREAD

**Issue by jfinkhaeuser** [bookwyrm-social/bookwyrm#2273]
**Is your feature request related to a problem? Please describe.**
This is a fairly advanced feature.

A lot of people I know have adopted fairly simple, but predictable order in which to sort physical book shelves. It would be nice to be able to replicate that on the software.

The order tends to be:
1. By last name, first name of author(s) in order listed on the book
2. By series order within authors. If the series changes authors, most often people tend to sort the series under whoever wrote the first entry.
3. By (first) publication date.

**Describe the solution you'd like**
Since the above is pretty specific, I think it'd be nice to have a simple query-language like interface for determining the sort order. One could imagine dragging the field names into an order or some such.

Have this editable in settings as a default order, but allow overriding the order in a per-shelf setting.

**Comment by edelgrace** [comment]
I think sorting by 1 (Last Name, First Name which  is what LibraryThing does) and 3 (first publication date) would be the easiest thing to implement here and can be tackled first.

I'm trying to imagine what the second request would look like. As far as I know, neither LibraryThing nor GoodReads has functionality like this. The simplest thing I can imagine is something like advanced sorting in spreadsheet editors. For example, in LibreOffice:

![2022-11-16_07-20](https://user-images.githubusercontent.com/3009729/202205178-fcd8a420-6981-4c22-be91-c563e641031d.png)

Any thoughts?

**Comment by jfinkhaeuser** [comment]
Yeah. That's also my ideal. Basically provide all data fields as possible options. Series number is already a field, after all, so I could build "my" sort order in this way.

**Comment by Skivling** [comment]
Could also be useful to be able to sort by the date started or finished reading.
